lib/RealWlanInterface.cpp

```diff
@@ -298,20 +298,29 @@ void RealWlanInterface::OnDisconnected(const WLAN_CONNECTION_NOTIFICATION_DATA&
     }
 }
 
-std::future<std::vector<ScannedBss>> RealWlanInterface::Scan(std::optional<const Ssid>& ssid)
+std::future<std::pair<std::vector<ScannedBss>, ScanStatus>> RealWlanInterface::Scan(std::optional<const Ssid>& ssid)
 {
+    std::unique_lock scanLock(m_promiseMutex);
+    if (m_scanRunning)
+    {
+        // A scan was already scheduled. Wait for its completion to provide results
+        Log::Trace(L"A scan was already scheduled on interface %ws. Waiting for its completion.", GuidToString(m_interfaceGuid).c_str());
```

Comment:
nit: some strange whitespace here

Reply:
Can you be more specific? Whitespaces look normal to me here.